homeassistant/components/decora_wifi/common.py

```diff
@@ -0,0 +1,179 @@
+        except ValueError as exc:
+            raise CommFailed from exc
+        self._logged_in = True
+
+    def _api_logout(self):
+        """Log out of decora_wifi session."""
+        if self._logged_in:
+            try:
+                Person.logout(self._session)
+            except ValueError as exc:
+                raise CommFailed from exc
+        self._logged_in = False
+
+    def _api_get_devices(self):
+        """Update the device library from the API."""
+
+        try:
+            # Gather all the available devices into the iot_switches dictionary...
+            perms = self._session.user.get_residential_permissions()
+
+            for permission in perms:
+                if permission.residentialAccountId is not None:
+                    acct = ResidentialAccount(
+                        self._session, permission.residentialAccountId
+                    )
+                    residences = acct.get_residences()
+                    for res in residences:
+                        switches = res.get_iot_switches()
+                        for switch in switches:
+                            # Add the switch to the appropriate list in the iot_switches dictionary.
+                            platform = DecoraWifiPlatform.classifydevice(switch)
+                            self._iot_switches[platform].append(switch)
+                elif permission.residenceId is not None:
+                    residence = Residence(self._session, permission.residenceId)
+                    switches = residence.get_iot_switches()
+                    for switch in switches:
+                        # Add the switch to the appropriate list in the iot_switches dictionary.
+                        platform = DecoraWifiPlatform.classifydevice(switch)
+                        self._iot_switches[platform].append(switch)
+        except ValueError as exc:
+            raise CommFailed from exc
+
+    def reauth(self):
+        """Reauthenticate this object's session."""
+        self._api_logout()
+        self._session = DecoraWiFiSession()
+        self._api_login()
+
+    def refresh_devices(self):
+        """Refresh this object's devices."""
+        self._iot_switches: dict[str, IotSwitch] = {
+            platform: [] for platform in PLATFORMS
+        }
+        self._api_get_devices()
+
+    def setup(self):
+        """Set up the session after object instantiation."""
+        self._api_login()
+        self._api_get_devices()
+
+    def teardown(self):
+        """Clean up the session on object deletion."""
+        self._api_logout()
+
+    @staticmethod
+    async def async_setup_decora_wifi(hass: HomeAssistant, email: str, password: str):
+        """Set up a decora wifi session."""
+
+        def setupplatform() -> DecoraWifiPlatform:
+            platform = DecoraWifiPlatform(email, password)
+            platform.setup()
+            return platform
+
+        return await hass.async_add_executor_job(setupplatform)
```

Comment:
```suggestion
        return await hass.async_add_executor_job(setup_platform)
```

Please use snake case to separate words